popmon/extensions/profile_diptest.py

```diff
@@ -0,0 +1,41 @@
+"""Hartigan & Hartigan's dip test for unimodality
+
+How to enable this extension:
+    - Install te diptest package: `pip install diptest` or `pip install popmon[diptest]`
+    - To show the diptest values in your report:
+        settings.report.show_stats.append("diptest*")
+        OR
+        settings.report.extended_report = True
+
+"""
+import numpy as np
+
+from popmon.analysis import Profiles
+from popmon.extensions.extension import Extension
+
+
+def extension():
+    from diptest import diptest
+
+    @Profiles.register(
+        key=["diptest_value", "diptest_pvalue"],
+        description=[
+            "diptest value for Hartigan & Hartigan's test for unimodality",
+            "p-value for the diptest",
+        ],
+        dim=1,
+        htype="num",
+    )
+    def diptest_profile(bin_centers, bin_values):
+        # unpack histogram into ordered samples
+        sample = np.repeat(bin_centers, bin_values.astype(int))
```

Comment:
I would suggest to add uniform noise to the samples, as diptest assumes a continuous distribution.
e.g.
```python
rng = np.random.default_rng()
hwb = (bin_centers[1] - bin_centers[0]) / 2
sample = np.repeat(bin_centers, bin_values.astype(int)) + rng.uniform(-hbw, hbw, bin_values.sum())
```